### PR TITLE
solution to 2 Convert a number to base n

### LINKED ISSUE #5: 2 - Convert a number to base n
Write a program to convert a number `N` from `base 10` to `base n`.

1. Generate a random number `n`.
2. Generate a random integer `N` in base 10.
3. Convert the integer `N` to `base n`, and output the result.

## Things to consider
1. Make sure the value of `n` is a positive integer between 2 and 16 (`n` should not be 10)
2. Make sure that `N` is a positive integer less than INFINITY

### PR DESCRIPTION
**Pull request by ebosetalee**
Resolves #5 [https://repl.it/join/uzyempke-emmanuellaesezo](https://repl.it/join/uzyempke-emmanuellaesezo) - join link
[https://repl.it/@EmmanuellaEsezo/2-Convert-a-number-to-base-n](https://repl.it/@EmmanuellaEsezo/2-Convert-a-number-to-base-n) 2nd link.

### CODE REVIEW

**Comment by jirevwe** [comment]
Your code outputted this `13 to base 16 is 31` 👀

**Comment by ebosetalee** [comment]
Fixed the issue and link to view[https://repl.it/join/uzyempke-emmanuellaesezo](https://repl.it/join/uzyempke-emmanuellaesezo)

File: convert-a-number-to-base-n.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      d = a // base_number
+      res+=str(c)
+      a = d
+    print("{0} to base {1:2} is {2}".format(number,base_number,res[::-1]))
+
+def calclulate_base_16():
+    print("{0} to base 16 is {0:>06x}".format(number))
+
+def base_11_to_15():
+    base_value = "0123456789abcde"
```
Comment (jirevwe) [inline convert-a-number-to-base-n.py]: The list is not complete, it should also be in uppercase
```suggestion
    base_value = "0123456789ABCDEF"
```

File: convert-a-number-to-base-n.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    a = number
+    while (a > 0):
+      c = a % base_number
+      d = a // base_number
+      res+=str(c)
+      a = d
+    print("{0} to base {1:2} is {2}".format(number,base_number,res[::-1]))
+
+def calclulate_base_16():
+    print("{0} to base 16 is {0:>06x}".format(number))
```
Comment (jirevwe) [inline convert-a-number-to-base-n.py]: The zero padding is nor needed. Are you sure you wrote this yourself? 👀 
```suggestion
    print("{0} to base 16 is {0:>x}".format(number))
```

File: convert-a-number-to-base-n.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    while (a > 0):
+      c = a % base_number
+      d = a // base_number
+      res+= base_value[c]
+      a = d
+    print("{0} to base {1} is {2}".format(number,base_number,res[::-1]))
+
+if base_number < 10:
+    calculate_base()
+elif (base_number == 16):
```
Comment (jirevwe) [inline convert-a-number-to-base-n.py]: All numbers above base 10 use the same notation that base 16 uses, so there's no need for the extra comparison.

**Review by jirevwe — CHANGES_REQUESTED** [review]

**Comment by ebosetalee** [comment]
Fixed the errors @jirevwe

**Review by jirevwe — APPROVED** [review]

**Review by yudori — APPROVED** [review]
Approving with a few notes since it's already late.

- Your base calculation would have been better off in one function. Both functions perform the same operations for the most part, they just have different input. Separating it has caused a lot of repetition.
- Make your indentations consistent. If you're using 4 spaces, stick to that throughout.
- Try to use more descriptive variable names

**Comment by ebosetalee** [comment]
Thank you sir @yudori

**Pull request merged**